models/issue.go

```diff
@@ -26,7 +26,7 @@ import (
 // Issue represents an issue or pull request of repository.
 type Issue struct {
 	ID               int64       `xorm:"pk autoincr"`
-	RepoID           int64       `xorm:"INDEX UNIQUE(repo_index)"`
+	RepoID           int64       `xorm:"UNIQUE(repo_index)"`
```

Comment:
`repo_id` index should not be removed.

Reply:
`IDX_issue_repo_id` is included in `UQE_issue_repo_index` as they have the same prefix.

`UQE_issue_repo_index` ==> `repo_id, index`
`IDX_issue_repo_id` ==> `repo_id`

Should I put it back anyway?

Reply:
No it is not needed

Reply:
I think maybe not all databases support the first index column on a united index as an index automatically?

Reply:
[This article](https://www.vividcortex.com/blog/the-left-prefix-index-rule) specifically mentions MySQL and PostgreSQL. Here's [one for SQL Server](https://aboutsqlserver.com/2010/10/20/composite-indexes/), and [this tutorial](https://www.sqlitetutorial.net/sqlite-index/) covers an example for SQLite.
It's a basic SQL feature, so I think we're safely covered. The only risk I see for this kind of optimization is _forgetting_ what indexes are used, needed and why.

Reply:
This should be supported on all DB types we use